ion-hash/src/type_qualifier.rs

```diff
@@ -7,11 +7,16 @@ use std::slice;
 ///! [spec]: https://amzn.github.io/ion-hash/docs/spec.html.
 use ion_rs::{
     binary::IonTypeCode,
-    value::{AnyInt, Element},
+    value::{AnyInt, Element, Sequence, Struct, SymbolToken},
     IonType,
 };
 use num_bigint::Sign;
 
+// For many types, the qualifier is either 'null' or 'not null'. That's what
+// these constants are for!
+const QUALIFIER_NOT_NULL: u8 = 0x00;
+const QUALIFIER_NULL: u8 = 0x0F;
```

Comment:
I think you need `0x01` for the `$0` case, also `bool` special cases these (`0x0` for `false` and `0x1` for `true`)

Reply:
Ok

Reply:
I made this change and don't like it. It doesn't enable code-reuse and I find the indirection annoying. Happy to actually do this, but I personally don't like it.

Reply:
Constants are useful to give things names (make intention clear) or for re-use. I don't think defining single-use constants here helps with either. The intentional is already clear.